### ISSUE TITLE
Vulnerabilities found in Minio Docker Image 

### ISSUE THREAD

**Issue by jithurjacob** [minio/minio#11216]
Hi Team, I've scanned the minio docker images with trivy and several vulnerabilities are found.

## Expected Behavior
Upon scanning with Trivy/Anchore or any other scanner no CVE's are expected in the ideal case.

## Current Behavior
92 vulnerabilities are discovered by Trivy.

minio/minio:edge (redhat 8.3)
Total: 92 (UNKNOWN: 0, LOW: 38, MEDIUM: 53, HIGH: 1, CRITICAL: 0)
Sample:
![image](https://user-images.githubusercontent.com/7101452/103524349-6b31e580-4ea3-11eb-818c-de90aba111a0.png)

## Possible Solution
Upgrade the base images to fix vulnerabilities.

## Steps to Reproduce (for bugs)
1. Install Trivy https://github.com/aquasecurity/trivy
2. Run Trivy scan with command - `trivy minio/minio:edge`

## Context
I'm working with an Enterprise customer that has strict security standards. We use multiple container scanners on all containers to ensure no vulnerabilities exist in any of the containers. We tried latest and edge images which are the most recent builds but CVE's are found in all.

## Regression
No

## Your Environment
* Version used (`minio --version`): minio/minio:edge docker image
* Server setup and configuration: Ubuntu
* Operating System and version (`uname -a`): Linux ip-172-31-26-217 5.4.0-1030-aws #31~18.04.1-Ubuntu SMP Tue Nov 17 10:48:34 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

**Comment by klauspost** [comment]
There are no issues reported on the upstream image: https://catalog.redhat.com/software/containers/ubi8-minimal/5c64772edd19c77a158ea216?container-tabs=security

So until that is updated there isn't much we can do.

Furthermore, none of the listed libraries are used by MinIO. It does not rely on any external code once running, so it shouldn't affect any running servers.

See https://github.com/minio/minio/blob/master/VULNERABILITY_REPORT.md#vulnerability-management-policy

If you would like a more detailed security review you can have a look at our [subnet](https://min.io/pricing) option.

**Comment by harshavardhana** [comment]
> There are no issues reported on the upstream image: https://catalog.redhat.com/software/containers/ubi8-minimal/5c64772edd19c77a158ea216?container-tabs=security
> 
> So until that is updated there isn't much we can do.
> 
> Furthermore, none of the listed libraries are used by MinIO. It does not rely on any external code once running, so it shouldn't affect any running servers.
> 
> See https://github.com/minio/minio/blob/master/VULNERABILITY_REPORT.md#vulnerability-management-policy
> 
> If you would like a more detailed security review you can have a look at our [subnet](https://min.io/pricing) option.

Closing this issue as there are no real reports here.

**Comment by jithurjacob** [comment]
@harshavardhana @klauspost  - Thank you for your quick response. I do not know how the upstream ubi8-minimal is tracking security issues. But when we scan the base image with `trivy registry.access.redhat.com/ubi8/ubi-minimal:8.3` it is showing the same 92 vulnerabilities. 

Although Minio may not be using these affected packages, in an Enterprise the CVE's against the containers are considered and we have a hard time explaining any exceptions to our customers. Is it possible for us to build our own image with a base image as alpine so that there is no CVE?

**Comment by harshavardhana** [comment]
> Although Minio may not be using these affected packages, in an Enterprise the CVE's against the containers are considered and we have a hard time explaining any exceptions to our customers. Is it possible for us to build our own image with a base image as alpine so that there is no CVE?

https://github.com/minio/minio/blob/master/VULNERABILITY_REPORT.md#vulnerability-management-policy you read our policy, you have to provide us full details on how you are affected.  Running a blind vulnerability check is not useful and there is nothing actionable for us. 

That doesn't mean you are affected, neither it means you are using such tools in the first place.

> @harshavardhana @klauspost - Thank you for your quick response. I do not know how the upstream ubi8-minimal is tracking security issues. But when we scan the base image with `trivy registry.access.redhat.com/ubi8/ubi-minimal:8.3` it is showing the same 92 vulnerabilities.

You should really report this to RedHat @jithurjacob we don't own the base image if you wish to have these vulnerabilities addressed, we cannot address them on our end. 

It's like running MinIO baremetal with host OS RHEL, MinIO is not responsible for these vulnerabilities. 

> Although Minio may not be using these affected packages, in an Enterprise the CVE's against the containers are considered and we have a hard time explaining any exceptions to our customers. Is it possible for us to build our own image with a base image as alpine so that there is no CVE?

You are free to do what you want what is available in public repositories. If you wish to become customers are you free to do as well where we can communicate this to RedHat through our internal channels.

**Comment by harshavardhana** [comment]
For example to double down on this look at the security reports in RedHat UBI images https://catalog.redhat.com/software/containers/ubi8/ubi-minimal/5c359a62bed8bd75a2c3fba8?container-tabs=security 

and the actual bugs in trivy tool which blindly reports random RPMs to be vulnerable 

https://github.com/aquasecurity/trivy/issues/749 

From what I can tell this is why we created vulnerability report guidelines where people should do their digging to understand what they are trying to report, blind scans are useless and there are random false positives throwing all over the place.

**Comment by harshavardhana** [comment]
Another example 

```
+------------------------+------------------+----------+--------------------+-------------------+-----------------------------------------+
| coreutils-single       | CVE-2017-18018   | MEDIUM   | 8.30-8.el8         |                   | coreutils: race condition               |
|                        |                  |          |                    |                   | vulnerability in chown and chgrp        |
|                        |                  |          |                    |                   | -->avd.aquasec.com/nvd/cve-2017-18018   |
```

Already addressed in the 

```
* Tue Jan 23 2018 Kamil Dudka <kdudka@redhat.com> - 8.29-2
- doc: warn about following symlinks recursively in chown/chgrp (CVE-2017-18018)
- mv -n: do not overwrite the destination
```

I don't even know how you can trust this tool there is no vulnerability reported as per RedHat and they are managing this for their 1000's of enterprise customers. I think trivy tool is not mature yet - the results should be treated with a grain of salt.

**Comment by harshavardhana** [comment]
Another one 
```
+------------------------+------------------+----------+--------------------+-------------------+-----------------------------------------+
|        LIBRARY         | VULNERABILITY ID | SEVERITY | INSTALLED VERSION  |   FIXED VERSION   |                  TITLE                  |
+------------------------+------------------+----------+--------------------+-------------------+-----------------------------------------+
| bash                   | CVE-2019-18276   | LOW      | 4.4.19-12.el8      |                   | bash: when effective UID is not         |
|                        |                  |          |                    |                   | equal to its real UID the...            |
|                        |                  |          |                    |                   | -->avd.aquasec.com/nvd/cve-2019-18276   |```
```

```
* Mon Jun 22 2020 Siteshwar Vashisht <svashisht@redhat.com> - 4.4.19-11
- Drop saved UID when effective UID is not equal to its real UID
  Resolves: #1793943
```

Already addressed here.

**Comment by jithurjacob** [comment]
@harshavardhana  - Thank you for the detailed response. I didn't get into the details of false positives given by Trivy for RedHat Images. I will go through them in detail.
